Web Server

Lnleite opened the pull request:
# Day 1 and 2
- [x] Implement send_response()
- [x] Examine handle_http_request() in the file server.c
- [x] Implement the get_d20() handler. This will call send_response()
- [x] Implement arbitrary file serving

# Day 3 and 4
- [x] Implement cache_put() in cache.c
- [x] Implement cache_get() in cache.c
- [x] Add caching functionality to server.c

# Stretch Goals
- [x] Automatic index.html serving

Lnleite committed:
Create notes.md to write down outline and thoughts

Lnleite committed:
Implement send_response()

Lnleite committed:
Notes for handle_http_request

Lnleite committed:
Examine handle_http_request() in the file server.c

Lnleite committed:
Start adding d20

RachelDiCesare93 commented on src/server.c:
```diff
@@ -52,37 +52,58 @@ int send_response(int fd, char *header, char *content_type, void *body, int cont
+    timestamp = localtime(&rawtime);
+    strftime(buffer, 100, "%a %b %d %T %Z %Y", timestamp);
     // Build HTTP response and store it in response
 
```
Looking good Lauren!

Lnleite committed:
Implement get_d20 and Add void for warnings

Lnleite committed:
Clear and edit comments

Lnleite committed:
Implement arbitrary file serving

Lnleite committed:
Update send response for the body

Lnleite committed:
Change response_length variable

- Used snprintf to get the length at the same time when building
the response.

RachelDiCesare93 commented on src/server.c:
```diff
@@ -132,39 +167,53 @@ void get_file(int fd, struct cache *cache, char *request_path)
+    }
 
     // (Stretch) If POST, handle the post request
 }
```
I highly encourage you to try stretch!

Lnleite committed:
Implement cache_put() in cache.c

Along with the cahce_put() being implemented:
- alloc_entry() functional
- free_entry() functional
- cache_create() functional

The tests are running as a pass for these functions

Lnleite committed:
Implement cache_get() in cache.c

Lnleite committed:
Add caching functionality to server.c

Lnleite committed:
Automatic index.html serving
